giskard_vision/detectors/base.py

```diff
@@ -101,30 +102,30 @@ def get_issues(
 
             from .example_manager import ImagesExampleManager
 
-            if issue_levels is None:
-                issue_levels = (IssueLevel.MAJOR, IssueLevel.MEDIUM)
-
-            for result in results:
-                if result.issue_level in issue_levels:
-                    issues.append(
-                        Issue(
-                            model,
-                            dataset,
-                            level=result.issue_level,
-                            slicing_fn=result.name,
-                            group=IssueGroup(
-                                result.group,
-                                self.warning_messages[result.group] if result.group in self.warning_messages else "",
-                            ),
-                            meta=result.get_meta_required(),
-                            scan_examples=ImagesExampleManager(result.filename_examples, embed=embed),
-                            display_footer_info=False,
-                        )
-                    )
-
         except (ImportError, ModuleNotFoundError) as e:
             raise GiskardImportError(["giskard"]) from e
 
+        if issue_levels is None:
+            issue_levels = (IssueLevel.MAJOR, IssueLevel.MEDIUM)
+
+        for result in results:
+            if result.issue_level in issue_levels:
+                issues.append(
+                    Issue(
+                        model,
+                        dataset,
+                        level=result.issue_level,
+                        slicing_fn=result.name,
+                        group=IssueGroup(
+                            result.group,
+                            self.warning_messages[result.group] if result.group in self.warning_messages else "",
```

Comment:
Yeah, I still don't like this, let's have the IssueGroups declared outside the detector class, like we do in https://github.com/Giskard-AI/giskard/blob/main/giskard/scanner/issues.py

Reply:
you can add `issue_group` attr to `ScanResult` and fill it while you create the issues inside each detector with the predefined issue groups, not on the level of the base detector.